File: Week_01/id_26/LeetCode_938_26.swift
class Solution {
    private var sums: Int = 0
    private var result: [Int] = []
    func rangeSumBST(_ root: TreeNode?, _ L: Int, _ R: Int) -> Int {
        inOrderBST(root)
        for (_, value) in result.enumerated() {
            if value >= L && value <= R {
                sums += value
            }
        }
        return sums
    }
    
    func inOrderBST(_ root: TreeNode?) {
        guard let root = root else {
            return
        }
        inOrderBST(root.left)
        result.append(root.val)
        inOrderBST(root.right)
    }
}

File: Week_02/id_26/LeetCode_938.26.py
class Solution(object):
    def rangeSumBST(self, root, L, R):
        def dfs(node):
            if node:
                if L <= node.val <= R:
                    self.ans += node.val
                    dfs(node.left)
                    dfs(node.right)
                if L > node.val:
                    dfs(node.right)
                if node.val > R:
                    dfs(node.left)

        self.ans = 0
        dfs(root)
        return self.ans